From 9977cbba8a36d3e6365cf886630d213ec7c1240c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Wed, 14 Jan 2015 16:29:24 +0000
Subject: [PATCH] classic draw rectangles "fit to contour" utterly broken

regression from

commit 3cbfcee36dd16e451d76c561eaaaee0ff29b01a5
Date:   Sun Feb 2 22:16:48 2014 -0600
    xub_StrLen and tools/string.hxx final straw

    There is still some 0xffff limit left and possibly some
    less than gracefully handled overflow/error cases

Change-Id: I82bdcb6951c42fdcaa68808a251f9b2f76f8ed0b
---
 editeng/source/editeng/impedit3.cxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/editeng/source/editeng/impedit3.cxx b/editeng/source/editeng/impedit3.cxx
index 494aafc..57cedf4 100644
--- a/editeng/source/editeng/impedit3.cxx
+++ b/editeng/source/editeng/impedit3.cxx
@@ -844,7 +844,7 @@ bool ImpEditEngine::CreateLines( sal_Int32 nPara, sal_uInt32 nStartPosY )
                 if ( pTextRanges->size() )
                 {
                     long nA = pTextRanges->at(0);
-                    long nB = pTextRanges->at(0);
+                    long nB = pTextRanges->at(1);
                     DBG_ASSERT( nA <= nB, "TextRange distorted?" );
                     long nW = nB - nA;
                     if ( nW > nMaxRangeWidth )
-- 
1.9.3

